modules/nixos/common/update.bash

```diff
@@ -6,25 +6,36 @@ if [[ "$(readlink /run/current-system)" == "$p" ]]; then
   exit 0
 fi
 
-nix-store --option narinfo-cache-negative-ttl 0 --realise "$p"
-nix-env --profile /nix/var/nix/profiles/system --set "$p"
+nix-store \
```

Comment:
I would like to add this code eventually to buildbot-nix in its current form (also with support for both macos & NixOS).

Reply:
macos support is complicated, nix-darwin doesn't have something like `restartIfChanged` for services and there isn't a way of monitoring failed services like we have on linux. I was planning to just keep using push deployments for darwin but switching from actions to effects.

Reply:
Would it be possible to come up with a common interface for both? Otherwise deploy tools all have to take care of those differences.

Reply:
Sorry, I don't understand what a "common interface" means here if it isn't a deployment tool?